### PR TITLE
Feat: Staking page Select

### PR DESCRIPTION
**Pull request by tuul-wq**
- Used new `Select` on Staking page
- Updated `Dropdown` so it can occupy width of the `Trigger`
- Removed `settingsStorage`
- Updated alignment of Select, Total staked, Total rewards

part of #2589 

image

### CODE REVIEW

File: src/renderer/shared/ui-kit/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -64,10 +64,15 @@ const Separator = () => {
   );
 };
 
-const Content = ({ children }: PropsWithChildren) => {
+type ContentProps = {
+  width?: 'auto' | 'trigger';
```
Comment (tuul-wq) [inline src/renderer/shared/ui-kit/Dropdown/Dropdown.tsx:68]: To have some control over `Content` width

File: src/renderer/shared/ui-kit/Select/Select.tsx [hunk 2]
```diff
@@ -19,18 +19,18 @@ type ContextProps = {
 
 const Context = createContext<ContextProps>({});
 
-type ControlledSelectProps = {
+type ControlledSelectProps<T extends string> = {
   placeholder: string;
-  value: string;
-  onChange: (value: string) => void;
+  value: T | undefined | null;
```
Comment (tuul-wq) [inline src/renderer/shared/ui-kit/Select/Select.tsx]: In order to avoid casting `ChainId` and other string types to `string`

Comment (johnthecat) [reply src/renderer/shared/ui-kit/Select/Select.tsx]: Why undefined and null at the same time? Let's get rid of undefined and and user will explicitly declare empty state

File: src/renderer/pages/Staking/ui/NetworkInfo.tsx [hunk 3]
```diff
@@ -36,77 +38,49 @@ export const NetworkInfo = ({
...
+          </Select>
         </div>
         {totalInfo.map(({ isLoading, title, amount, asset }) =>
-          isLoading || !asset ? (
+          isLoading || nullable(asset) ? (
             <div key={title} className="flex flex-col gap-y-1" data-testid="value-loading">
               <FootnoteText className="text-text-secondary">{title}</FootnoteText>
-              <Shimmering width={122} height={20} />
-              <Shimmering width={47} height={18} />
+              <Skeleton width="120px" height="16px" />
```
Comment (johnthecat) [inline src/renderer/pages/Staking/ui/NetworkInfo.tsx]:
```suggestion
              <Skeleton width={30} height={4} />
```

File: src/renderer/pages/Staking/ui/NetworkInfo.tsx [hunk 4]
```diff
@@ -36,77 +38,49 @@ export const NetworkInfo = ({
...
+              <Skeleton width="40px" height="16px" />
```
Comment (johnthecat) [inline src/renderer/pages/Staking/ui/NetworkInfo.tsx]:
```suggestion
              <Skeleton width={10} height={4} />
```

**Review by johnthecat — APPROVED** [review]

**Pull request merged**